### PR TITLE
Avoid shema comparison to None

### LINKED ISSUE #54: Segmentation fault when running pytest on OSX
Pytest breaks when we run kartothek on OSX Mojave

I have followed the steps mentioned in the CONTRIBUTING.md.

It breaks with `Segmentation fault: 11` when running `test_eq` module in the `test_metapartition.py`.

Here the pytest logs:

```
(kartothek-dev) ganesh:kartothek 10251XX$ pytest -v -k test_metapartition.py
=================================================================================================== test session starts ====================================================================================================
platform darwin -- Python 3.6.6, pytest-4.5.0, py-1.8.0, pluggy-0.12.0 -- /Users/10251XX/open_kartothek/kartothek-dev/bin/python3.6
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/10251XX/open_kartothek/kartothek/.hypothesis/examples')
rootdir: /Users/10251XX/open_kartothek/kartothek, inifile: setup.cfg
plugins: hypothesis-4.23.9, cov-2.7.1, mock-1.10.4, flake8-1.0.4
collected 1524 items / 1455 deselected / 69 selected                                                                                                                                                                       

tests/io_components/test_metapartition.py::test_store_single_dataframe_as_partition_no_metadata[4] PASSED                                                                                                            [  1%]
tests/io_components/test_metapartition.py::test_store_multiple_dataframes_as_partition[json-4] PASSED                                                                                                                [  2%]
tests/io_components/test_metapartition.py::test_store_single_dataframe_as_partition[json-4-dataset_uuid/core/test_label.parquet] PASSED                                                                              [  4%]
tests/io_components/test_metapartition.py::test_store_multiple_dataframes_as_partition[msgpack-4] PASSED                                                                                                             [  5%]
tests/io_components/test_metapartition.py::test_store_single_dataframe_as_partition[msgpack-4-dataset_uuid/core/test_label.parquet] PASSED                                                                           [  7%]
tests/io_components/test_metapartition.py::test_load_dataframes[4-True] PASSED                                                                                                                                       [  8%]
tests/io_components/test_metapartition.py::test_load_dataframes[4-False] PASSED                                                                                                                                      [ 10%]
tests/io_components/test_metapartition.py::test_remove_dataframes[4] PASSED                                                                                                                                          [ 11%]
tests/io_components/test_metapartition.py::test_load_dataframes_selective[4] PASSED                                                                                                                                  [ 13%]
tests/io_components/test_metapartition.py::test_load_dataframes_columns_projection[4] PASSED                                                                                                                         [ 14%]
tests/io_components/test_metapartition.py::test_load_dataframes_columns_raises_missing[4] PASSED                                                                                                                     [ 15%]
tests/io_components/test_metapartition.py::test_to_dict[4] PASSED                                                                                                                                                    [ 17%]
tests/io_components/test_metapartition.py::test_reconstruct_date_index[4-True] PASSED                                                                                                                                [ 18%]
tests/io_components/test_metapartition.py::test_reconstruct_date_index[4-False] PASSED                                                                                                                               [ 20%]
tests/io_components/test_metapartition.py::test_column_string_cast[4] PASSED                                                                                                                                         [ 21%]
tests/io_components/test_metapartition.py::test_from_dict PASSED                                                                                                                                                     [ 23%]
tests/io_components/test_metapartition.py::test_eq Segmentation fault: 11
(kartothek-dev) ganesh:kartothek 10251XX$ 
```

Here the details of the system:
```
Darwin ganesh.local 18.2.0 Darwin Kernel Version 18.2.0: Thu Dec 20 20:46:53 PST 2018; root:xnu-4903.241.1~1/RELEASE_X86_64 x86_64
```

Here the details of packages installed:

```
alabaster==0.7.12
appnope==0.1.0
aspy.yaml==1.3.0
asv==0.4
atomicwrites==1.3.0
attrs==19.1.0
Babel==2.7.0
backcall==0.1.0
certifi==2019.3.9
cfgv==2.0.0
chardet==3.0.4
Click==7.0
cloudpickle==1.1.1
coverage==4.5.3
dask==1.2.2
decorator==4.4.0
distributed==1.28.1
docutils==0.14
entrypoints==0.3
flake8==3.7.7
flake8-mutable==1.2.0
HeapDict==1.0.0
hypothesis==4.23.9
identify==1.4.3
idna==2.8
imagesize==1.1.0
importlib-metadata==0.15
importlib-resources==1.0.2
ipython==7.5.0
ipython-genutils==0.2.0
jedi==0.13.3
Jinja2==2.10.1
-e 
locket==0.2.0
MarkupSafe==1.1.1
mccabe==0.6.1
more-itertools==7.0.0
msgpack==0.6.1
nodeenv==1.3.3
numpy==1.16.4
packaging==19.0
pandas==0.24.2
parso==0.4.0
partd==0.3.10
pexpect==4.7.0
pickleshare==0.7.5
pluggy==0.12.0
pre-commit==1.16.1
prompt-toolkit==2.0.9
psutil==5.6.2
ptyprocess==0.6.0
py==1.8.0
pyarrow==0.13.0
pycodestyle==2.5.0
pyflakes==2.1.1
Pygments==2.4.2
pyparsing==2.4.0
pytest==4.5.0
pytest-cov==2.7.1
pytest-flake8==1.0.4
pytest-mock==1.10.4
python-dateutil==2.8.0
pytz==2019.1
PyYAML==5.1
requests==2.22.0
setuptools-scm==3.3.3
simplejson==3.16.0
simplekv==0.12.0
six==1.12.0
snowballstemmer==1.2.1
sortedcontainers==2.1.0
Sphinx==2.0.1
sphinx-rtd-theme==0.4.3
sphinxcontrib-applehelp==1.0.1
sphinxcontrib-devhelp==1.0.1
sphinxcontrib-htmlhelp==1.0.2
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.2
sphinxcontrib-serializinghtml==1.1.3
storefact==0.8.4
tblib==1.4.0
toml==0.10.0
toolz==0.9.0
tornado==6.0.2
traitlets==4.3.2
uritools==2.2.0
urllib3==1.25.3
virtualenv==16.6.0
wcwidth==0.1.7
zict==0.1.4
zipp==0.5.1
zstandard==0.11.1
```

Thanks,
Ganesh

### PR DESCRIPTION
**Pull request by fjetter**
# Description:

Fixes the segfault cause by https://issues.apache.org/jira/browse/ARROW-5873

- [x] Closes #54
- [x] Tests added / passed
- [x] Passes `./format_code.sh`
- [x] Changelog entry

### CODE REVIEW

File: kartothek/io_components/metapartition.py [hunk 1]
```diff
@@ -340,7 +340,11 @@ def __eq__(self, other):
             return False
 
         for table, meta in self.table_meta.items():
-            if not meta.equals(other.table_meta.get(table, None)):
+            # https://issues.apache.org/jira/browse/ARROW-5873
```
Comment (lr4d) [inline kartothek/io_components/metapartition.py]: Can you add a comment saying this only affects pyarrow versions <= 0.14.0 ?

Comment (fjetter) [reply kartothek/io_components/metapartition.py]: That info is available on the JIRA ticket ;)

File: CHANGES.rst [hunk 2]
```diff
@@ -2,8 +2,10 @@
 Changelog
 =========
 
-Version 3.2.1 (2019-XX-XX)
-==========================
+Next version (Unreleased)
+=========================
+
+- Fix an issue where a segmentation fault may be raised when comparing MetaPartition instances
```
Comment (crepererum) [inline CHANGES.rst]: FYI: the issue is fixed in pyarrow 0.14.1 which is out now. I would blacklist 0.14.0 and only allow 0.14.1, so the description needs to be adjusted here.

Comment (fjetter) [reply CHANGES.rst]: I don't why we would need to adjust this. There is no arrow 0.14.1 compatible Kartothek version

Comment (fjetter) [reply CHANGES.rst]: So, I'm just wondering what the changelog should read :) I don't have a strong opinion here

Comment (crepererum) [reply CHANGES.rst]: If we don't support `0.14.0` and will just finish #108 to support `0.14.1` (which still blacklists `0.14.0`), I think from a user perspective "Add support for pyarrow 0.14.0" would be a valid changelog entry.

Comment (fjetter) [reply CHANGES.rst]: This isn't about arrow versions. The issue is fixed in 0.14.1. but still exists in all other versions. The reason why we don't want to support 0.14.0 is a completely different topic and has nothing to do with this segfault

Comment (crepererum) [reply CHANGES.rst]: Ah I remember, this is actually also buggy in older versions, right? Then replace `segfault` with `potential segfault` and then this should be OK.

**Comment by fjetter** [comment]
I changed the OSX build job setting to disallow failure for OSX from now on. If it makes trouble we can open the discussion again

**Pull request merged**